frontend/app/src/App.tsx

```diff
@@ -512,6 +526,16 @@ export class App extends PureComponent<Props, State> {
       window.prerenderReady = true
     }
     if (this.state.scriptRunState !== prevState.scriptRunState) {
+      mark(this.state.scriptRunState)
+
+      if (this.state.scriptRunState === ScriptRunState.NOT_RUNNING) {
```

Comment:
As written, it will run in production but should not have any impact outside of adding metadata for any browser's devtools. I don't personally see an issue with having it in all builds, in fact it may even be helpful if attempting to debug something in a prod build, but let me know if you feel otherwise.

Reply:
Yeah if we don't expect any performance impact or any permission issues (e.g. on SiS) then I think it should be fine to leave it in 👍  If it requires any special privileges or has to load third-party stuff that can be blocked by a CSP we should be careful (I don't think it does though)

Reply:
That's a great call out, I appreciate you bringing that up. Just confirming that it does not require any special privileges and is built-in to the browser built-in Performance API.